### PR TITLE
49490_task_first_commet

### PR DESCRIPTION
**Pull request by dina28**

### CODE REVIEW

File: fels_100/categoriesViewController.m [hunk 1]
```diff
@@ -30,11 +47,14 @@ - (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
...
 
 - (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
-    return 5;
+    return categoriesName.count;
 }
 
 - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
     categoriesTableViewCell *cell= [tableView dequeueReusableCellWithIdentifier:NSStringFromClass([categoriesTableViewCell class]) forIndexPath:indexPath];
+    NSDictionary *theDic=[categoriesName objectAtIndex: indexPath.row ];
+    cell.categoriesTypeLb.text=[theDic objectForKey:@"name"];
```
Comment (nguyenvandungb) [inline fels_100/categoriesViewController.m]: line 55,56 add space " = "

File: fels_100/categoriesViewController.m [hunk 2]
```diff
@@ -8,17 +8,34 @@
...
 
 
 @interface categoriesViewController ()
 
 @end
 
-@implementation categoriesViewController
-  
+@implementation categoriesViewController {
+    NSDictionary * Dictionary;
```
Comment (nguyenvandungb) [inline fels_100/categoriesViewController.m]: You define variable here. It is not wrong but better if you move to line 13,14

**Comment by nguyenvandungb** [comment]
OK

File: fels_100/DataAccess.m [hunk 3]
```diff
@@ -33,4 +35,18 @@ -(void)signUp:(NSString *)name email:(NSString *)email password:(NSString *)pass
     }];
 }
 
+-(void )getCategories:(NSString *) nothing complete:(void (^)(NSDictionary* check))completionBlock {
+    NSDictionary *param = @{ @"page": @1,
```
Comment (TequilarTomsk) [inline fels_100/DataAccess.m]: Why the page and auth_token is fixed here?

File: fels_100/categoriesViewController.m [hunk 4]
```diff
@@ -8,17 +8,31 @@
...
+
 - (void)viewDidLoad {
     [super viewDidLoad];
     [self.categoriesTableView registerNib:[UINib nibWithNibName:NSStringFromClass([categoriesTableViewCell class] ) bundle:nil] forCellReuseIdentifier:NSStringFromClass([categoriesTableViewCell class])];
+    [self loadCategories];
+}
+
+-(void)loadCategories {
+    DataAccess *access = [[DataAccess alloc]init];
+    [access getCategories:@"nothing"
```
Comment (TequilarTomsk) [inline fels_100/categoriesViewController.m]: why do you put string "nothing" here

File: fels_100/categoriesViewController.m [hunk 5]
```diff
@@ -8,17 +8,31 @@
...
+                     page:@2
```
Comment (TequilarTomsk) [inline fels_100/categoriesViewController.m]: Why page is 2?

File: fels_100/DataAccess.h [hunk 6]
```diff
@@ -15,4 +15,6 @@
 
 - (void)signUp:(NSString *)name email:(NSString *)email password:(NSString *)password confirm:(NSString *)confirmPassword complete:(void(^)(BOOL isLoggedIn))completionBlock;
 - (void)signIn:(NSString *)email password:(NSString *)password remember:(NSNumber *)rememberMe complete:(void(^)(BOOL isLogged,NSDictionary *theDic))completionBlock;
+- (void )getCategories:(NSNumber*)page auto_token:(NSString*)auto_token complete:(void (^)(BOOL check ,NSDictionary* categoriesDictionary))completionBlock;
```
Comment (TequilarTomsk) [inline fels_100/DataAccess.h]: what is the auto_token?

File: fels_100/DataAccess.m [hunk 7]
```diff
@@ -57,4 +57,15 @@ - (void)signIn:(NSString *)email password:(NSString *)password remember:(NSNumbe
...
 }
 
+-(void )getCategories:(NSNumber*)page auto_token:(NSString*)auto_token complete:(void (^)(bool check ,NSDictionary* categoriesdict))completionBlock {
+    NSDictionary *param = @{ @"page":page,
+                             @"auth_token":auto_token};
+    [[self getManager] GET:@"categories.json" parameters:param progress:nil
+        success:^( NSURLSessionTask *task, id responseObject) {
+            completionBlock(YES, responseObject);
+        } failure:^(NSURLSessionTask *operation, NSError *error) {
+            completionBlock(NO ,@{}.mutableCopy);
```
Comment (TequilarTomsk) [inline fels_100/DataAccess.m]: mutable copy for what in this case??

File: fels_100/categoriesViewController.m [hunk 8]
```diff
@@ -8,17 +8,63 @@
...
+
+- (void)handleRefresh:(UIRefreshControl *)refreshControl {
+    [self loadCategories];
+    [refreshControl endRefreshing];
+}
+
+-(void)getPageNumber{
+    DataAccess *access = [[DataAccess alloc]init];
+    [access getCategories:@1
+               auto_token:@"nCVjGJZZQDx-uvenYiwQ0w"
```
Comment (TequilarTomsk) [inline fels_100/categoriesViewController.m]: why token is hardcode here??

File: fels_100/categoriesViewController.m [hunk 9]
```diff
@@ -8,17 +8,63 @@
...
+-(void)loadCategories {
+    int k = [self.currentPage integerValue];
+    k = (k+1)%[self.totalPage integerValue];
+    if(k == 0) {
+        k = [self.totalPage integerValue];
+    }
+    self.currentPage = [NSNumber numberWithInt:k];
+        DataAccess *access = [[DataAccess alloc]init];
+        [access getCategories:self.currentPage
+                auto_token:@"nCVjGJZZQDx-uvenYiwQ0w"
```
Comment (TequilarTomsk) [inline fels_100/categoriesViewController.m]: token is hard code here????

File: fels_100/categoriesViewController.m [hunk 10]
```diff
@@ -8,17 +8,63 @@
...
+    [refreshController addTarget:self action: @selector(handleRefresh:) forControlEvents:UIControlEventValueChanged];
+    [self.categoriesTableView addSubview:refreshController];
+}
+
+- (void)handleRefresh:(UIRefreshControl *)refreshControl {
+    [self loadCategories];
+    [refreshControl endRefreshing];
+}
+
+-(void)getPageNumber{
```
Comment (TequilarTomsk) [inline fels_100/categoriesViewController.m]: Why do you call function [access getCategories] two time for getting information???? It's not good logic and decrease the performance

**Pull request merged**